Debug dmdrfe

joshua-cogliati-inl opened the pull request:
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)


##### What are the significant changes in functionality due to this change request?
Creative pythoning, to try and fix some pickling problems.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [ ] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

moosebuild commented:
Job [Mingw Test](https://civet.inl.gov/job/2836571/) on 3e3de47 : invalidated by @joshua-cogliati-inl

failed in fetch Directory not empty

joshua-cogliati-inl commented:
Mischief managed.^H^H Debugging done.

joshua-cogliati-inl commented:
Also, if anyone is debugging dask pickle errors, I used the following code in tokenize.py in Dask to get the error:
```python
def _normalize_pickle(o: object) -> tuple:
    buffers: list[pickle.PickleBuffer] = []
    pik: int | None = None
    pik2: int
    success = False
    for mod in [pickle, cloudpickle]:
        if success:
            break
        for _ in range(3):
            buffers.clear()
            try:
                out = mod.dumps(o, protocol=5, buffer_callback=buffers.append)
                mod.loads(out, buffers=buffers)
                pik2 = hash_buffer_hex(out)
            except Exception as e:
                print("npf",repr(e),"from",o)
                break
            if pik == pik2:
                success = True
                break
            pik = pik2
        else:
            _maybe_raise_nondeterministic("Failed to tokenize deterministically")
            break
    if pik is None:
        _maybe_raise_nondeterministic("Failed to tokenize deterministically")
        pik = int(uuid.uuid4())
    return pik, [hash_buffer_hex(buf) for buf in buffers]
```
Specifically, I added the line `print("npf",repr(e),"from",o)` and added the `as e`.  Note that also some of the changes in 033a5a1d2e5b85921eadc9f801a9dab2b5a517c3 can be helpful.
